AntGame/src/antgame/Match.java

```diff
@@ -114,169 +114,253 @@ private void step(Ant ant) throws Exception {
+                    if (currentCell.containsFood()) {
+                        if (!ant.hasFood()) {
+                            currentCell.decrementFoodInCell();
+                            ant.setHasFood(true);
+                            nextState = pickup.getTrueState();
+                        } else {
+                            nextState = pickup.getFalseState();
+                        }
                     } else {
-                        nextState = sense.getFalseState();
+                        nextState = pickup.getFalseState();
                     }
-                } else {
-                    nextState = sense.getFalseState();
-                }
-            } else if (sense.getCond().getCondition().equals(ConditionType.FoeWithFood)) {
-                if (cellOccupiedBy < 0) {
-                    nextState = sense.getFalseState();
-                } else if (ants[cellOccupiedBy].getColour().equals(foeColour)) {
-                    if (ants[cellOccupiedBy].hasFood()) {
-                        nextState = sense.getTrueState();
+                } else if (instToExec instanceof Drop) {
+                    Drop drop = (Drop) instToExec;
+                    if (ant.hasFood()) {
+                        currentCell.incrementFoodInCell();
+                        ant.setHasFood(false);
+                        nextState = drop.getStateToGoTo();
                     } else {
-                        nextState = sense.getFalseState();
+                        nextState = drop.getStateToGoTo();
+                    }
+                } else if (instToExec instanceof Turn) {
+                    Turn turn = (Turn) instToExec;
+                    if (turn.getTurnDirection().equals(TurnDirection.Left)) {
+                        int newDirection = (ant.getDirection() + 5) % 6;
+                        ant.setDirection(newDirection);
+                        nextState = turn.getStateToGoTo();
+                    } else if (turn.getTurnDirection().equals(TurnDirection.Right)) {
+                        int newDirection = (ant.getDirection() + 1) % 6;
+                        ant.setDirection(newDirection);
+                        nextState = turn.getStateToGoTo();
+                    } else {
+                        throw new Exception("Can only turn left or right");
+                    }
+                } else if (instToExec instanceof Move) {
+                    Move move = (Move) instToExec;
+                    Cell cellToMoveTo = board.getCellAtCurrentPositionPlusDirection(ant.getCurrentPosition(), ant.getDirection(), SenseDirection.Ahead);
+                    if (cellToMoveTo.getTerrain().equals(Terrain.ROCKY) | cellToMoveTo.getOccupier() > -1) {
+                        nextState = move.getStateToGoToIfBlocked();
+                    } else {
+                        currentCell.removeOccupation();
+                        cellToMoveTo.setOccupied(ant.getId());
```

Comment:
Need 
  currentCell.removeAntColour(); //reset colour
  cellToMoveTo.setAntColour(ant.getColour()); //change colour

to remove previous colour and set next cell to allow ant rendering